extensions/wikia/DataProvider/DataProvider.php

```diff
@@ -427,47 +429,78 @@ public static function GetNewlyCreatedArticles($limit = 5, $ns = array(NS_MAIN))
 
 	/**
 	 * Return array of top five users
+	 *
+	 * Called by NavigationModel::handleExtraWords
+	 *
 	 * Author: Inez Korczynski (inez at wikia.com)
 	 * @return array
 	 */
 	final public static function GetTopFiveUsers($limit = 7) {
 		wfProfileIn(__METHOD__);
-		global $wgStatsDB, $wgMemc, $wgCityId, $wgStatsDBEnabled;
+		global $wgStatsDB, $wgStatsDBEnabled;
+
+		$fname = __METHOD__;
+		$limit = min( $limit, self::TOP_USERS_MAX_LIMIT );
 
 		if (empty($wgStatsDB) || empty($wgStatsDBEnabled)) {
-			$result = array();
+			$results = [];
 		}
 		else {
-			$memckey = wfMemcKey("TopFiveUsers", $limit);
-			$results = $wgMemc->get($memckey);
+			$results = WikiaDataAccess::cacheWithLock( wfMemcKey( __CLASS__, 'GetTopFiveUsers' ), WikiaResponse::CACHE_STANDARD, function() use ($fname) {
+				global $wgCityId, $wgStatsDB;
 
-			if (!is_array($results)) {
 				$dbr = wfGetDB(DB_SLAVE);
-				$row = $dbr->selectRow("user_groups", "GROUP_CONCAT(ug_user) AS user_list", array("ug_group IN ('staff', 'bot')"), __METHOD__);
-
-				$dbs = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
-				$query = "SELECT user_id AS rev_user, edits AS cnt FROM specials.events_local_users WHERE wiki_id = '" . $wgCityId . "' " . (!empty($row->user_list) ? "AND user_id NOT IN (" . $row->user_list . ",'0','929702')" : "") . " ORDER BY edits DESC";
-
-				$res = $dbs->query($dbs->limitResult($query, $limit * 4, 0));
-
-				$results = array();
+				$users_list = $dbr->selectFieldValues(
+					"user_groups",
+					"ug_user",
+					[ 'ug_group' => [ 'staff', 'bot' ] ],
+					$fname
+				);
+
+				// add more blacklisted user IDs
+				$users_list[] = 0;
+				$users_list[] = 22439; // Wikia
+				$users_list[] = 929702; // CreateWiki script
```

Comment:
Good point, but I'm afraid that we will end up with completely rewritten code here :)